src/amqp.c

```diff
@@ -86,6 +86,7 @@ const char * const QD_AMQP_COND_RESOURCE_DELETED = "amqp:resource-deleted";
 const char * const QD_AMQP_COND_ILLEGAL_STATE = "amqp:illegal-state";
 const char * const QD_AMQP_COND_FRAME_SIZE_TOO_SMALL = "amqp:frame-size-too-small";
 const char * const QD_AMQP_COND_CONNECTION_FORCED = "amqp:connection:forced";
+const char * const QD_AMQP_COND_MESSAGE_SIZE_EXCEEDED = "amqp:connection:message-size-exceeded";
```

Comment:
Not sure if this matters, but AMQP 1.0 defines a link-error for message-size-exceeded, not a connection-error:

[AMQP 1.0 link-error](http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-transport-v1.0-os.html#type-link-error)

Reply:
Good point. Change the error to
    condition name: amqp:connection:forced
    condition description: message size exceeded

Reply:
Why close the connection rather than the link?

Reply:
Earlier versions of this feature tried closing the link with an error. This did not work well with any of the proton clients, in particular the python client used by dispatch self tests. 

See https://issues.apache.org/jira/browse/PROTON-2182

Reply:
My tests tried only the proton-c clients. I didn't try any proton-j clients. Sorry for the mischaracterization of "any of the proton clients".

Reply:
To be clearer, I was talking about the sending link, vs sending connection. In this case the example having both the sender and receiver of the message on the same connection probably confuses matters a touch.

From the error and stacktrace that appears to be more on the receiver side. Perhaps a general issue with handling aborted deliveries? Unclear it relates to choice of whether to close the sending link vs connection, other than the fact both are on the same connection.

Reply:
Its arguably not that big a deal to the sending client, as anyone sending over the limit is going to have issues either way.

I was more curious given the commentry seems to suggest inter-router connections could be getting killed too in certain cases, which could affect other things besides the one doing the send? Or is that not the case? Though I guess that effect may still be true of link closure too.